modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -266,6 +266,26 @@ inline fun <A, B> EitherOf<A, B>.getOrHandle(default: (A) -> B): B =
 inline fun <A, B> EitherOf<A, B>.filterOrElse(predicate: (B) -> Boolean, default: () -> A): Either<A, B> =
   flatMap { if (predicate(it)) Right(it) else Left(default()) }
 
+/**
+ * * Returns [Either.Right] with the existing value of [Either.Right] if this is a [Either.Right] and the given
+ * predicate holds for the right value.
+ * * Returns `Left(default({right}))` if this is a [Either.Right] and the given predicate does not
+ * hold for the right value. Useful for error handling where 'default' returns a message with context on why the value
+ * did not pass the filter
+ * * Returns [Either.Left] with the existing value of [Either.Left] if this is a [Either.Left].
+ *
+ * Example:
+ * ```
```

Comment:
We should use here a Kotlin ank executable doc snippet so it's type checked and runnable in the docs . See the docs module for examples

Reply:
Oops. Obviously didn't read the comment well enough.

Updated to reflect function it represents, with an example that returns Left and uses the Right's value.

Reply:
What Raul is saying is that the docs tooling we use can execute the results of these snippets, rather than us hardcoding them. See for example:

https://github.com/arrow-kt/arrow/edit/master/modules/docs/arrow-docs/docs/docs/datatypes/option/README.md

What you can do is split this snippet in two sub-snippets, each starting with ```kotlin:ank

Reply:
I was actually referring to the executable ones not just regular ank snippets. https://github.com/arrow-kt/arrow/blob/master/modules/effects/arrow-effects/src/main/kotlin/arrow/effects/typeclasses/Bracket.kt#L40-L72

Reply:
Ok. I'll take a look at the examples, figure out what's required, and update accordingly